CHT/Temperature.C

```diff
@@ -21,10 +21,27 @@ void preciceAdapter::CHT::Temperature::write(double* buffer, bool meshConnectivi
 
     if (this->locationType_ == LocationType::volume)
     {
-        forAll(T_->ref(), i)
+        if(cellSetNames_.empty())
         {
-            buffer[bufferIndex++] = T_->ref()[i];
+            forAll(T_->ref(), i)
+            {
+                buffer[bufferIndex++] = T_->ref()[i];
+            }
+        }
+        else
+        {
+            for (uint j = 0; j < cellSetNames_.size(); j++) {
+                cellSet overlapRegion(T_->mesh(), cellSetNames_[j]);
+	            const labelList & cells = overlapRegion.toc();
+
+                for( int i=0; i < cells.size(); i++)
+                {
+                    // Copy temperature into the buffer
+                    buffer[bufferIndex++] = T_->ref()[cells[i]];
+                }
```

Comment:
Could we simplify by using a ranged for loop here?
This question should probably also refer to the `forAll` in the empty cellSet case.